cmd/main.go

```diff
@@ -1,8 +1,128 @@
 package main
 
-import "fmt"
+import (
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/rs/cors"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/rog-golang-buddies/rmx/service"
+)
+
+// var (
+// 	port = flag.Int("SERVER_PORT", 8888, "The port that the server will be running on")
+// )
 
 func main() {
-	// Feel free to delete this file.
-	fmt.Println("Hello Gophers")
+	if err := initCLI().Run(os.Args); err != nil {
+		log.Fatalln(err)
+	}
 }
+
+func run(cfg *Config) error {
+	sCtx, cancel := signal.NotifyContext(context.Background(), syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	defer cancel()
+
+	// ? should this defined within the instantiation of a new service
+	c := cors.Options{
+		AllowedOrigins:   []string{"*"}, // ? band-aid, needs to change to a flag
+		AllowCredentials: true,
+		AllowedMethods:   []string{http.MethodGet, http.MethodPost},
+		AllowedHeaders:   []string{"Origin", "Content-Type", "Accept", "Authorization"},
+	}
+
+	srv := http.Server{
+		Addr:    ":" + strconv.Itoa(cfg.Port),
+		Handler: cors.New(c).Handler(service.New(chi.NewMux())),
+		// max time to read request from the client
+		ReadTimeout: 10 * time.Second,
+		// max time to write response to the client
+		WriteTimeout: 10 * time.Second,
+		// max time for connections using TCP Keep-Alive
+		IdleTimeout: 120 * time.Second,
+		BaseContext: func(_ net.Listener) context.Context { return sCtx },
+		ErrorLog:    log.Default(),
+	}
+
+	g, gCtx := errgroup.WithContext(sCtx)
+
+	g.Go(func() error {
+		// Run the server
+		srv.ErrorLog.Printf("App server starting on %s", srv.Addr)
+		return srv.ListenAndServe()
+	})
+
+	g.Go(func() error {
+		<-gCtx.Done()
+		return srv.Shutdown(context.Background())
+	})
+
+	// if err := g.Wait(); err != nil {
+	// 	log.Printf("exit reason: %s \n", err)
+	// }
+
+	return g.Wait()
+}
+
+// func getEnv(key, fallback string) string {
```

Comment:
Let's delete all these comments if we don't need the code anymore.

A buddy of mine had this suggestion about committing commented-out code:
> Leave a comment describing the conditions needed for the code to be deleted or comment back in. If you don't know what those conditions are, delete the code.

This way we have the confidence to delete these chunks